server/src/main/java/com/cloud/network/element/ConfigDriveNetworkElement.java

```diff
@@ -573,6 +575,10 @@ private boolean deleteConfigDriveIsoOnHostCache(final VirtualMachine vm, final L
             LOG.warn(String.format("Host %s appears to be unavailable, skipping deletion of config-drive ISO on host cache", hostId));
             return false;
         }
+        if (!Arrays.asList(Status.Up, Status.Connecting).contains(hostVO.getStatus())) {
+            LOG.warn(String.format("Host status %s is not Up or Connecting, skipping deletion of config-drive ISO on host cache", hostId));
```

Comment:
@weizhouapache @vishesh92 Would `connecting` be included, asking as what if the host isn't up to take the command?

Reply:
@rohityadavcloud This returns false if the status is NOT `Up` or `Connecting`..
I matching the checks which are present here: https://github.com/apache/cloudstack/blob/4.18/engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java#L970